src/commands/org/open.ts

```diff
@@ -130,7 +130,16 @@ export class OrgOpenCommand extends SfCommand<OrgOpenOutput> {
 
     // create a local html file that contains the POST stuff.
     const tempFilePath = path.join(tmpdir(), `org-open-${new Date().valueOf()}.html`);
-    await fs.promises.writeFile(tempFilePath, getFileContents(conn.accessToken as string, conn.instanceUrl, retUrl));
+    await fs.promises.writeFile(
+      tempFilePath,
+      getFileContents(
+        conn.accessToken as string,
+        conn.instanceUrl,
+        // the path flag is URI-encoded in its `parse` func.
+        // For the form redirect to work we need it decoded.
+        flags.path ? decodeURIComponent(flags.path) : retUrl
```

Comment:
`retUrl` can be either a generated path from `--source-file` or `--path`.
If the former that's fine because it doesn't get encoded in `generateFileUrl`, so we only check for `flags.path`.